### PR TITLE
Add Odoo 17.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
- Dockerfile copied from 16.0 and slightly adapted
- Fix tests
- Make it `:latest`.
- Deprecate some old Odoo versions.
- Bump PG to 15.

TODO: Switch to Python 3.10

@Tecnativa TT45808

### CODE REVIEW

File: 17.0.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,232 @@
+FROM python:3.8-slim-bullseye AS base
```
Comment (celm1990) [inline 17.0.Dockerfile:1]: Odoo 17 need minimum  PY 3.0 according this commit
https://github.com/odoo/odoo/commit/163fb3ae618bf993e60f8c1276e09a5b85565fa5
Do you sure this working with 3.8?

Comment (pedrobaeza) [reply 17.0.Dockerfile:1]: I see, we will need to make it work with P3.10. I was delaying such step, as it was conflicting for 16 (and still not working).

Comment (celm1990) [reply 17.0.Dockerfile:1]: I tested into commit https://github.com/celm1990/doodba/commit/888f30b6fca66b54daba167a6f3917d89e68f324 and it's working now

**Comment by pedrobaeza** [comment]
Reopening other PR for avoiding obsolete branches marked as required.

**Pull request closed without merging**